src/components/atoms/InputField/InputField.tsx

```diff
@@ -0,0 +1,58 @@
+import React, { forwardRef } from "react";
+import classNames from "classnames";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { IconProp } from "@fortawesome/fontawesome-svg-core";
+import { isTruthy } from "utils/types";
+import "./InputField.scss";
+
+export interface InputFieldProps extends React.HTMLProps<HTMLInputElement> {
+  iconStart?: IconProp | JSX.Element | false;
+  iconEnd?: IconProp | JSX.Element | false;
+}
```

Comment:
```suggestion
export interface InputFieldProps extends React.HTMLProps<HTMLInputElement> {
  iconStart?: IconProp | JSX.Element;
  iconEnd?: IconProp | JSX.Element;
}
```

Why do we need to have false here?

Reply:
I included `false` when I started to use `InputField` in `NavSearchBar`. 

```tsx
<InputField
...
  iconEnd={
    isTruthy(searchQuery) && (
      <img
        className="nav__clear-search"
        src="/icons/nav-dropdown-close.png"
        alt="close button"
        onClick={clearSearchQuery}
      />
    )
  }
/>  
```

I supposed that it'd be helpful to have `false` for such cases. If we remove it from icon type alias we'll have to write something like this: 

```tsx
<InputField
    iconEnd={
      isTruthy(searchQuery) ?
        <img
          className="nav__clear-search"
          src="/icons/nav-dropdown-close.png"
          alt="close button"
          onClick={clearSearchQuery}
        />
      : undefined
    }
  />
```

But sure, we can go without it, if you think that it's not needed. or maybe you have another idea how to pass this props in a better way?

Reply:
I see.
Let's stick to the second option `isTruthy ? icon : undefined`. it looks a bit more explicit

Reply:
updated here: https://github.com/sparkletown/sparkle/pull/1306/commits/13040f8da655ed6b3a8ed7e889e8411f1d9da6f3

Reply:
Is JSX.Element the correct ‘catch all’ type for ‘things that can be rendered’, or is there a better one? I remember there was one that handles everything, but can’t remember it’s name.